includes/functions.php

```diff
@@ -1497,6 +1547,33 @@ function tsml_import_buffer_set($meetings, $data_source_url = null, $data_source
     $indexes_to_remove = [];
 
     for ($i = 0; $i < count($meetings); $i++) {
+
+        //figure out a singular slug from available sources, starting with 'slug' (of course)
+        $meeting_slug = '';
+        foreach (array('slug','post_name', 'name', 'title', 'group', 'location') as $field) {
+            if (!$meeting_slug) { 
+                $meeting_slug = isset($meetings[$i][$field]) ? sanitize_title($meetings[$i][$field]) : '';
+            }
+        }
+
+        //if we have no slug at this point, something went wrong - remove the meeting and continue
+        if (!$meeting_slug) {
+            array_splice($meetings, $i, 1);
+            --$i;
+            continue;
+        }
+
+        //ensure every slug is unique for this import list
+        if (isset($meeting_slugs[$meeting_slug])) {
+            while (isset($meeting_slugs[$meeting_slug . '-' . $meeting_slugs[$meeting_slug]]) && $meeting_slugs[$meeting_slug] < 999) {
+                $meeting_slugs[$meeting_slug]++;
+            }
+            $meeting_slug = $meeting_slug . '-' . $meeting_slugs[$meeting_slug]++;
+        } else {
+            $meeting_slugs[$meeting_slug] = 1;
+        }
+        $meetings[$i]['slug'] = $meeting_slug;
+
```

Comment:
cool! so the logic here is that it tries to use the above series of fields, and sanitizes then and adds a number 1-999 to slugify the record? 

not sure if we're doing this already but GSO instructed some google sheets to use a field called `ID` in lieu of slug, so we should support that

also `post_name` is a field that wordpress uses in their db but i don't think we expect that in JSON, CSV, or in a sheet

Reply:
Fair, likely if `post_name` exists, `slug` would as well. I was largely doing that for overkill - just exhaustively trying to find all valid fields. I'll add `ID`, but will reject it if it's a number like a WordPress post ID.